tests/archive/test_archive.py

```diff
@@ -30,7 +30,7 @@
         param("sample.txt", "sample.txt", id="uncompressed file"),
     ],
 )
-def test_open_gz_file(data_dir, src_file, expected_file) -> None:
```

Comment:
I might be wrong but should this be in the same format as 'StrPath' ?

Reply:
I have thought about it, but the parametrisations for now are only `str`, so to avoid an extra import I have kept it that way. If you believe it may make it easier to set the type as `StrPath`, happy to change it.

Reply:
I think its fine for now, Str should be the correct representation then.